csharp/ColumnDescriptor.cs

```diff
@@ -69,11 +69,22 @@ public TReturn Apply<TReturn>(LogicalTypeFactory typeFactory, Type? columnLogica
             var (physicalType, logicalType) = typeFactory.GetSystemTypes(this, columnLogicalTypeOverride);
             var elementType = logicalType;
 
-            for (var node = SchemaNode; node != null; node = node.Parent)
+            var node = SchemaNode;
+            while (node != null)
             {
-                if (node.LogicalType.Type == LogicalTypeEnum.List)
+                try
```

Comment:
I don't think that `try/finally` improves anything here. In my opinion it just adds complexity to the code without any real benefit.

Reply:
So you mean just remove the try/finally and keep the rest of the code as-is, so if there's an exception we let the finalizers run to free the handles? That seems reasonable to me, especially as this code is highly unlikely to throw.

Reply:
Do you think I should similarly remove `try/finally` elsewhere, eg. in the tests?

Reply:
Right, good point, I've changed this now.